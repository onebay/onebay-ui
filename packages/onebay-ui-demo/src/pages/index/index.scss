.page {
  position: relative;
  background-color: #f8f8f8;

  div {
    box-sizing: border-box;
  }
}

.page-index {
  padding-top: 30px;
  padding-bottom: 50px;

  .logo {
    margin: 0 auto;
    margin-top: 60px;
    font-size: 0;
    text-align: center;
  }
	.img {
		width: 264px;
		height: 180px;
	}
  .page-title {
    margin-top: 24px;
    color: #333;
    font-size: 18px;
    text-align: center;
  }

  .module-list {
    margin-top: 72px;
    display: flex;
    flex-direction: column;
    align-items: center;

    &__item {
      box-sizing: border-box;
      display: flex;
      flex-direction: row;
      align-items: center;
      margin-bottom: 20px;
      padding: 0 15px;
      width: 90%;
      height: 72px;
      background: #fff;
      box-shadow: 0 4px 20px 0 rgba(0, 0, 0, 0.04);
      border-radius: 5px;

      &:last-child {
        margin-bottom: 0;
      }
    }

    &__icon {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 35px;
      height: 35px;
      color: #78A4FA;
      text-align: center;
      border: 2px solid rgba(120, 164, 250, 0.16);
      border-radius: 50%;

      .img {
        width: 15px;
        height: 15px;
      }
    }

    &__info {
      flex: 1;
      text-align: left;
      margin-left: 24px;

      .title {
        color: #6A6A77;
        font-size: 14px;
        font-weight: bold;
      }

      .content {
        color: #88889C;
        font-size: 11px;
      }
    }

    &__arrow {
      display: flex;
      justify-content: center;
      align-items: center;
      width: 31px;
      height: 31px;
      color: #fff;
      text-align: center;
      background: #78A4FA;
      box-shadow: 2px 5px 15px 0 rgba(120, 164, 250, 0.30);
      border-radius: 50%;
    }
  }
}

.page-body {
  padding: 40px;
}

.component-group {
  font-size: 32px;
}

.group-item {
  padding: 0 30px;
  margin: 20px 0;
  background-color: #fff;
  border-radius: 4px;
  overflow: hidden;

  &:first-child {
    margin-top: 0;
  }
}

.group-info {
  padding: 30px 0;
  display: flex;
  align-items: center;
  transition: opacity 0.3s;

  &-title {
    opacity: 0.5;
  }
}

.group-list {
  font-size: 28px;

  .list-component {
    padding: 20px 0;
    position: relative;
    align-items: center;

    &::before {
      content: " ";
      position: absolute;
      left: 0;
      top: 0;
      right: 0;
      height: 1px;
      border-top: 1px solid #d8d8d8;
      color: #d8d8d8;
    }

    &:first-child::before {
      display: none;
    }

    &-info {
      width: 100%;
    }

    &-arrow {
      display: inline-block;
      height: 18px;
      width: 18px;
      border-width: 2px 2px 0 0;
      border-color: #888;
      border-style: solid;
      transform: matrix(0.71, 0.71, -0.71, 0.71, 0, 0) translate(-50%);
      position: absolute;
      top: 50%;
      right: 0;
    }
  }
}
