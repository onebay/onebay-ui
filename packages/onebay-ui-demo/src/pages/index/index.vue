<template>
  <div class="page page-index">
    <div class="logo">
      <img class="img" :src="taroLogo" />
    </div>
    <div class="page-title">onebay UI</div>
    <div class="module-list">
      <div
        v-for="(item, index) in list"
        :key="index"
        class="module-list__item"
        @click="gotoPanel(item.id)"
      >
        <div class="module-list__icon">
          <img :src="item.icon" class="img" mode="widthFix" />
        </div>
        <div class="module-list__info">
          <div class="title">
            {{ item.title }}
          </div>
          <div class="content">
            {{ item.content }}
          </div>
        </div>
        <div class="module-list__arrow">
          <span class="ob-icon ob-icon-chevron-right" />
        </div>
      </div>
    </div>
  </div>
</template>
<script>
import { defineComponent } from 'vue'
import iconAction from '../../assets/images/icon-list-action.png'
import iconBasic from '../../assets/images/icon-list-basic.png'
import iconForm from '../../assets/images/icon-list-form.png'
import iconLayout from '../../assets/images/icon-list-layout.png'
import iconNavigation from '../../assets/images/icon-list-navigation.png'
import iconView from '../../assets/images/icon-list-view.png'
import taroLogo from '../../assets/images/logo_onebay.png'
import './index.scss'

const list = [
  {
    id: 'Basic',
    title: 'Basic',
    content: 'Include Icon & Button',
    icon: iconBasic
  },
  {
    id: 'View',
    title: 'View',
    content: 'Include Noticebar, Tag, Badge, etc.',
    icon: iconView
  },
  {
    id: 'Action',
    title: 'Action',
    content: 'Include Modal, Progress, Toast, etc.',
    icon: iconAction
  },
  {
    id: 'Form',
    title: 'Form',
    content: 'Include Input, Radio, Checkbox, etc.',
    icon: iconForm
  },
  {
    id: 'Layout',
    title: 'Layout',
    content: 'Include Flex, List, Accordion, etc.',
    icon: iconLayout
  },
  {
    id: 'Navigation',
    title: 'Navigation',
    content: 'Include NavBar, Tabs, Drawer, etc.',
    icon: iconNavigation
  }
]

export default defineComponent({
  data() {
    return {
      taroLogo,
      list
    }
  },
  methods: {
    gotoPanel(id) {
      this.$router.push(`/pages/panel/index?id=${id.toLowerCase()}`)
    }
  }
})
</script>
