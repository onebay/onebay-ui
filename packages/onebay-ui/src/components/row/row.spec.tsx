import { mount } from '@vue/test-utils'
import { Row, Col } from '../../index'

describe('Row Snap', () => {
  it('render initial Row', () => {
    const wrapper = mount(<Row />)
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Row with Col', () => {
    const wrapper = mount(
      <Row>
        <Col>column 1</Col>
        <Col>column 2</Col>
      </Row>,
      {
        components: { Col }
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })
  it('render Row props gutter', () => {
    const wrapper = mount(
      <Row gutter={20}>
        <Col>column 1</Col>
        <Col>column 2</Col>
      </Row>,
      {
        components: { Col }
      }
    )
    expect(wrapper.element).toMatchSnapshot()
  })
})
describe('Row Event', () => {
  it('emit onClick', () => {
    const wrapper = mount(<Row />)
    wrapper.trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
  })
})