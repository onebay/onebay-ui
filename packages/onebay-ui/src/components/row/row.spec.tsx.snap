// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Snap render Row props gutter 1`] = `
<div
  class="ob-row"
>
  
  <div
    class="ob-col ob-col--1"
    style="padding-right: 10px;"
  >
    
    column 1
    
  </div>
  <div
    class="ob-col ob-col--1"
  >
    
    column 2
    
  </div>
  
</div>
`;

exports[`Row Snap render Row with Col 1`] = `
<div
  class="ob-row"
>
  
  <div
    class="ob-col ob-col--1"
  >
    
    column 1
    
  </div>
  <div
    class="ob-col ob-col--1"
  >
    
    column 2
    
  </div>
  
</div>
`;

exports[`Row Snap render initial Row 1`] = `
<div
  class="ob-row"
>
  <!---->
</div>
`;
